client/src/components/overview/QuantitySelector.jsx

```diff
@@ -40,34 +66,57 @@ const QuantitySelector = (props) => {
     return quantities;
   }
 
-  const selectAttributes = getAttributes(quantitiesAvailable);
-  const quantities = getQuantities(quantitiesAvailable);
-  if (quantitiesAvailable) {
-    resetToDefaultSelected();
+  onChange(e) {
+    const newQuantityValue = e?.target?.value;
+    const newQuantity = parseInt(newQuantityValue);
+    if (newQuantity) {
+      console.log('Quantity onChange:', newQuantity);
+      this.setState({ currentQuantity: newQuantity})
+      this.props.onSelect(newQuantity);
+    }
   }
 
-  let key = 0;
-  return (
-    <div id="po-quantity-selector">
-      <select
-        id="po-quantity"
-        name="quantity"
-        { ...selectAttributes }
-      >
-        {
-          quantitiesAvailable ? (
-            quantities.map(quantity => (
-              <option key={ key++ } value={ quantity }>
-                  { quantity }
-              </option>
-            ))
-          ) : (
-          <option value={ -1 }>-</option>
-          )
-        }
-      </select>
-    </div>
-  );
-};
+  render() {
+    console.log('Rendering Quantity Selector');
+
+    const quantitiesAreAvailable = this.quantitiesAreAvailable(this.props.maxQuantity, this.props.maxQuantityLimit);
+    console.log('quantitiesAreAvailable', quantitiesAreAvailable);
```

Comment:
This looks like a re-hash of stuff that's been handled earlier. Why not have class variables that take these values when they are first determined (e.g., `this.quantitiesAreAvailable`, `this.currentQuantityIndex`, etc.)?

Reply:
Yeah, this is one of those things for me to figure out. It needs to be checked at 2-3 different points in the app depending on circumstances (initialization, rendering, after render), where I need to make sure it updates.

I used to use class variables more, but you cannot make those read-only, so I'm torn between having alterable variables vs. making sure the values are always up to date with the latest props. Something to resolve during my learning of React patterns.